app/app-controllers/cart/cartModalController.js

```diff
@@ -93,6 +95,10 @@ function(
     };
 
     CartModalController.prototype.hide = function() {
+        if (this.errorController.isShowing) {
```

Comment:
Is it possible to remove the knowledge about the `errorController` here then in the errorController wait for it's hide to complete before calling the `backHandler`? Ex from thinkgeek: https://github.com/mobify/thinkgeek-ios/blob/develop/app/thinkgeek-controllers/connectivityController.js#L91